src/core/CMakeLists.txt

```diff
@@ -3,6 +3,8 @@
 
 SET(QGIS_CORE_SRCS
   ${CMAKE_SOURCE_DIR}/external/kdbush/include/kdbush.hpp
+  ${CMAKE_SOURCE_DIR}/external/nlohmann/json.hpp
+  ${CMAKE_SOURCE_DIR}/external/nlohmann/json_fwd.hpp
```

Comment:
Do these files need to be compiled? I think they are included anyway so no reason to put them into the SRCS here (and not sure about the kdbush either @nyalldawson )

Instead the external/CMakeFiles.txt  should probably include the logic to install them onto the system since they are de facto no longer "external" for QGIS - apart from code organization inside the source tree.

Reply:
Although kdbush is a bit different, it's currently only used in a private header (which is not deployed on an installation), so that does not need to be shipped.